plugin/pkg/scheduler/schedulercache/node_info.go

```diff
@@ -153,6 +165,18 @@ func calculateResource(pod *api.Pod) (cpu int64, mem int64, non0_cpu int64, non0
 	return
 }
 
+// Sets the overall node information.
+func (n *NodeInfo) SetNode(node *api.Node) error {
+	n.node = node
+	return nil
+}
+
+// Removes the overall information about the node.
+func (n *NodeInfo) RemoveNode(node *api.Node) error {
```

Comment:
I didn't do that on purpose - potentially there may be some race, that in the system pods are removed and then node is removed, but since pods & nodes are delivered in different watch channels, there may be a race and node deletion may be delivered first. In that case, we don't want to remove it until all pods are also removed. This is handled in cache.go